osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs

```diff
@@ -250,13 +272,13 @@ protected override void Update()
 
             timingPoint = BeatSyncSource.ControlPoints.TimingPointAt(BeatSyncSource.Clock.CurrentTime);
 
-            if (beatLength != timingPoint.BeatLength)
+            if (updateDivisor() || beatLength != timingPoint.BeatLength)
```

Comment:
this is turboweird. can we not have magic state-mutating method calls like this if we can avoid? maybe the method can be pure and just return the desired divisor?